services/packagist/packagist-dependency-version.service.js

```diff
@@ -0,0 +1,204 @@
+import Joi from 'joi'
+import { optionalUrl } from '../validators.js'
+import { NotFound } from '../index.js'
+import {
+  allVersionsSchema,
+  BasePackagistService,
+  customServerDocumentationFragment,
+} from './packagist-base.js'
+
+const queryParamSchema = Joi.object({
+  server: optionalUrl,
+  version: Joi.string(),
+}).required()
+
+export default class PackagistDependencyVersion extends BasePackagistService {
+  static category = 'platform-support'
+
+  static route = {
+    base: 'packagist/dependency-v',
+    pattern: ':user/:repo/:dependency+',
+    queryParamSchema,
+  }
+
+  static examples = [
+    {
+      title: 'Packagist Dependency Version Support',
+      namedParams: {
+        user: 'symfony',
+        repo: 'symfony',
+        dependency: 'twig/twig',
+      },
+      staticPreview: this.render({
+        dependency: 'twig/twig',
+        dependencyVersion: '2.13|^3.0.4',
+      }),
+    },
+    {
+      title: 'Packagist Dependency Version Support (specify version)',
+      namedParams: {
+        user: 'symfony',
+        repo: 'symfony',
+        dependency: 'twig/twig',
+      },
+      queryParams: {
+        version: 'v2.8.0',
+      },
+      staticPreview: this.render({
+        dependency: 'twig/twig',
+        dependencyVersion: '1.12',
+      }),
+    },
+    {
+      title: 'Packagist Dependency Version Support (custom server)',
+      namedParams: {
+        user: 'symfony',
+        repo: 'symfony',
+        dependency: 'twig/twig',
+      },
+      queryParams: {
+        server: 'https://packagist.org',
+      },
+      staticPreview: this.render({
+        dependency: 'twig/twig',
+        dependencyVersion: '2.13|^3.0.4',
+      }),
+      documentation: customServerDocumentationFragment,
+    },
+    {
+      title: 'Packagist PHP Version Support (custom server)',
+      namedParams: {
+        user: 'symfony',
+        repo: 'symfony',
+        dependency: 'php',
+      },
+      queryParams: {
+        server: 'https://packagist.org',
+        version: 'v2.8.0',
+      },
```

Comment:
I think I'd probably lose the params (server and version on this one)

Reply:
ok, done